samples/nrf9160/http_application_update/src/main.c

```
@@ -0,0 +1,246 @@
+							dfu->fragment_size);
+		flash_write_protection_set(flash_dev, true);
+		if (err != 0) {
+			printk("Flash write error %d at address %08x\n",
+				err, flash_address);
+			return 1;
+		}
+		flash_address += dfu->fragment_size;
+		break;
+	}
+
+	case DOWNLOAD_CLIENT_EVT_DOWNLOAD_DONE:
+		printk("DOWNLOAD_CLIENT_EVT_DOWNLOAD_DONE");
+		break;
+
+	case DOWNLOAD_CLIENT_EVT_ERROR: {
+		download_client_disconnect(dfu);
+		printk("Download client error, please restart "
+			"the application\n");
+		break;
+	}
+	default:
+		break;
+	}
+	return 0;
+}
+
+
+/**@brief Turn on LED0 and LED1 if CONFIG_APPLICATION_VERSION
+ * is 2 and LED0 otherwise.
+ */
+static void led_app_version(void)
+{
+	struct device *dev;
+
+	dev = device_get_binding(LED_PORT);
+	if (dev == 0) {
+		printk("Nordic nRF GPIO driver was not found!\n");
+		return;
+	}
+
+	gpio_pin_configure(dev, LED0_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(dev, LED0_GPIO_PIN, 1);
+
+#if CONFIG_APPLICATION_VERSION == 2
+	gpio_pin_configure(dev, LED1_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(dev, LED1_GPIO_PIN, 1);
+#endif
+}
+
+void dfu_button_pressed(struct device *gpiob, struct gpio_callback *cb,
+		    u32_t pins)
+{
+	gpio_pin_disable_callback(gpiob, SW0_GPIO_PIN);
+	start_dfu = true;
+}
+
+static void dfu_button_init(void)
+{
+	int err;
+
+	gpiob = device_get_binding(SW0_GPIO_CONTROLLER);
+	if (gpiob == 0) {
+		printk("Nordic nRF GPIO driver was not found!\n");
+		return;
+	}
+	err = gpio_pin_configure(gpiob, SW0_GPIO_PIN,
+		GPIO_DIR_IN		|
+		GPIO_INT		|
+		GPIO_PUD_PULL_UP|
+		GPIO_INT_EDGE	|
+		GPIO_INT_ACTIVE_LOW);
+	if (err == 0) {
+		gpio_init_callback(&gpio_cb, dfu_button_pressed,
+			BIT(SW0_GPIO_PIN));
+		err = gpio_add_callback(gpiob, &gpio_cb);
+	}
+	if (err == 0) {
+		err = gpio_pin_enable_callback(gpiob, SW0_GPIO_PIN);
+	}
+	if (err != 0) {
+		printk("Unable to configure SW0_GPIO_PIN!\n");
+	}
+}
+
+static void application_update(void)
+{
+	start_dfu = false;
+	app_dfu_init();
```

Please return an error code and verify it - if there's an error during initialization or during connection, there no use proceeding.

---

Fixed now.